src/commands/force/source/delete.ts

```diff
@@ -0,0 +1,206 @@
+    this.validateFlags();
+
+    this.componentSet = await ComponentSetBuilder.build({
+      apiversion: this.getFlag<string>('apiversion'),
+      sourceapiversion: await this.getSourceApiVersion(),
+      sourcepath: this.getFlag<string[]>('sourcepath'),
+      metadata: this.flags.metadata && {
+        metadataEntries: this.getFlag<string[]>('metadata'),
+        directoryPaths: this.getPackageDirs(),
+      },
+    });
+
+    this.sourceComponents = this.componentSet.getSourceComponents().toArray();
+
+    if (!this.sourceComponents.length) {
+      // if we didn't find any components to delete, let the user know and exit
+      // matches toolbelt
+      this.ux.styledHeader(chalk.blue('Deleted Source'));
+      this.ux.log('No results found');
+      this.exit(0);
+    }
+
+    // create a new ComponentSet and mark everything for deletion
+    const cs = new ComponentSet([]);
+    this.sourceComponents.map((component) => {
+      cs.add(component, true);
+    });
+    this.componentSet = cs;
+
+    if (!(await this.handlePrompt())) {
+      // the user said 'no' to the prompt
+      this.exit(0);
+    }
+
+    // fire predeploy event for the delete
+    await this.lifecycle.emit('predeploy', this.componentSet.toArray());
+    this.isRest = await this.isRestDeploy();
+    this.ux.log(`*** Deleting with ${this.isRest ? 'REST' : 'SOAP'} API ***`);
+
+    const deploy = await this.componentSet.deploy({
+      usernameOrConnection: this.org.getUsername(),
+      apiOptions: {
+        rest: this.isRest,
+        checkOnly: this.getFlag<boolean>('checkonly', false),
+        testLevel: this.getFlag<TestLevel>('testlevel'),
+      },
+    });
+    this.updateDeployId(deploy.id);
+
+    if (!this.isJsonOutput()) {
+      const progressFormatter: ProgressFormatter = env.getBoolean('SFDX_USE_PROGRESS_BAR', true)
+        ? new DeployProgressBarFormatter(this.logger, this.ux)
+        : new DeployProgressStatusFormatter(this.logger, this.ux);
+      progressFormatter.progress(deploy);
+    }
+
+    this.deployResult = await deploy.pollStatus(500, this.getFlag<Duration>('wait').seconds);
+    await this.lifecycle.emit('postdeploy', this.deployResult);
+  }
+
+  /**
+   * Checks the response status to determine whether the delete was successful.
+   */
+  protected resolveSuccess(): void {
+    const status = getString(this.deployResult, 'response.status');
+    if (status !== RequestStatus.Succeeded) {
+      this.setExitCode(1);
+    }
+  }
+
+  protected formatResult(): DeployCommandResult {
+    const formatterOptions = {
+      verbose: this.getFlag<boolean>('verbose', false),
+    };
+
+    const formatter = new DeleteResultFormatter(this.logger, this.ux, formatterOptions, this.deployResult);
+
+    // The DeleteResultFormatter will use SDR and scan the directory, if the files have been deleted, it will throw an error
+    // so we'll delete the files locally now
+    this.deleteFilesLocally();
```

Comment:
This feels out of place in a results formatting function, and I don't think the comment applies anymore.  What if we move this to either the `resolveSuccess` method or the `run` method  just after calling `resolveSuccess()`?